File: frontend/src/services/ProductService.js
import axios from "axios";
import { API_BASE_URL } from "./apiConfig";

const PRODUCT_API_URL = `${API_BASE_URL}/products`;

export const getAllProducts = async () => {
    const response = await axios.get(PRODUCT_API_URL);
    return response.data;
};

export const searchProducts = async (keyword) => {
    const response = await axios.get(`${PRODUCT_API_URL}/search`, {
        params: { keyword }
    });
    return response.data;
};

export const getProductById = async (productId) => {
    const response = await axios.get(`${PRODUCT_API_URL}/${productId}`);
    return response.data;
};

export const getAdminProductById = async (productId) => {
    const response = await axios.get(`${API_BASE_URL}/admin/products/${productId}`);
    return response.data;
};

export const getProductsBySeller = async (sellerId) => {
    const response = await axios.get(`${PRODUCT_API_URL}/seller/${sellerId}`);
    return response.data;
};

export const createProduct = async (productData) => {
    const response = await axios.post(PRODUCT_API_URL, productData);
    return response.data;
};

export const updateProduct = async (productId, productData) => {
    const response = await axios.put(`${PRODUCT_API_URL}/${productId}`, productData);
    return response.data;
};

export const deleteProduct = async (productId) => {
    await axios.delete(`${PRODUCT_API_URL}/${productId}`);
};

export const uploadProductImage = async (file) => {
    const formData = new FormData();

    formData.append("file", file);

    const response = await axios.post(
        `${API_BASE_URL}/upload`,
        formData
    );

    return response.data;
};

File: src/main/java/com/ebizzness/ecommerce/service/ProductService.java
package com.ebizzness.ecommerce.service;

import com.ebizzness.ecommerce.dto.request.ProductRequest;
import com.ebizzness.ecommerce.dto.response.ProductResponse;
import java.util.List;

public interface ProductService {
    ProductResponse createProduct(ProductRequest request);
    List<ProductResponse> getAllProducts();
    ProductResponse getProductById(Long id);
    ProductResponse getProductByIdForAdmin(Long id);
    ProductResponse updateProduct(Long id, ProductRequest request);
    void deleteProduct(Long id);
    List<ProductResponse> searchProducts(String keyword, String category, String courseCode, String status);
    List<ProductResponse> getProductsBySeller(Long sellerId);
}
